### DISCUSSION TITLE
how to use pyG to do graph level prediction

### DISCUSSION THREAD

**Discussion by lingxuan9** [pyg-team/pytorch_geometric#6807] [Q&A] [answered]
Hi!

Thanks for such amazing tool! 

I have a use case:  

There is a graph with 3000 edges,  both nodes and edges have some features. When removing a edge from the graph, a score got. I would like to predict such score(related to number of nodes, shortest paths & other features).

How can I perform PyG to a graph level prediction? It seems a graph level prediction problem.

Thanks for any advice.

**Comment by wsad1** [comment]
You'll probably need to use some sort of pooling layer to pool node/edge features. See an example of using topk pool to make graph level predictions on a protein [here](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/proteins_topk_pool.py). You can find all pooling layers in pyg [here](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#pooling-layers)

**Comment by lingxuan9** [comment, marked as answer]
Thanks for your reply @wsad1 . Let me describe the question a bit more.

Assume that there is a graph with m nodes, n edges, every time when removing a or serval edge(s) from the graph, a defined score is calculated. Now I would like to predict the value of the score when removing a/some new edges from the graph.

My solution: convert this question into a graph level prediction question. After removing a/some edges from the graph, actually we can get/generate a new graph. So we can get a dataset contains a lot of graphs by random removing edges from the graph. Each graph is corresponding to a score -> predict label. Now the issue is predict such score for a given graph.

My questions:
- Does the solution good? Is there other solutions? Is there other similar use case?
- Can pyG perform such issue well?

Thanks!

**Reply by rusty1s** [reply]
Yes, I think this is exactly how you would approach this.

**Reply by lingxuan9** [reply]
> Yes, I think this is exactly how you would approach this.

Thanks for your reply!

**Reply by lingxuan9** [reply]
Hi @rusty1s  sorry for bothering you. Is there any graph level regression example I can follow? I found this graph classification [example](https://colab.research.google.com/drive/1I8a0DfQ3fI7Njc62__mVXUlcAleUclnb?usp=sharing#scrollTo=mHSP6-RBOqCE), but would like to have more infos.  Thanks!

**Reply by rusty1s** [reply]
You can take a look here: https://github.com/pyg-team/pytorch_geometric/blob/master/examples/pna.py
